File: src/server/packets/PacketS2CZonePower.js
const PacketS2C = require("./PacketS2C");

/**
 * Server -> Client
 * ID = 0x08
 * Zone Power
 * Sends a zone's power status
 * Data:
 *      (Unsigned Char) Controller ID
 *      (Unisgned Char) Zone ID
 *      (Unsigned Char) On/Off
 */
class PacketS2CZonePower extends PacketS2C {
    constructor(ctrlrID, zoneID, power) {
        super();

        this._buffer.writeUInt8(ctrlrID);
        this._buffer.writeUInt8(zoneID);
        this._buffer.writeUInt8(power ? 0x01 : 0x00);
    }

    getID() {
        return 0x08;
    }
}

module.exports = PacketS2CZonePower;

File: src/RNetPi.Core/Packets/PacketS2CZonePower.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Server -> Client
/// ID = 0x08
/// Zone Power
/// Sends a zone's current power state
/// Data:
///     (Unsigned Char) Controller ID
///     (Unsigned Char) Zone ID
///     (Unsigned Char) Power State
/// </summary>
public class PacketS2CZonePower : PacketS2C
{
    public const byte ID = 0x08;

    public PacketS2CZonePower(byte controllerID, byte zoneID, bool powered)
    {
        Writer.Write(controllerID);
        Writer.Write(zoneID);
        Writer.Write((byte)(powered ? 1 : 0));
    }

    public override byte GetID() => ID;
}